File: 2540-minimum-common-value/2540-minimum-common-value.cpp
class Solution {
public:
    int getCommon(vector<int>& nums1, vector<int>& nums2) {
        int n = nums1.size();
        int m = nums2.size();

        int a = 0;
        int b = 0;

        while(a < n && b < m){
            if(nums1[a] == nums2[b]) return nums1[a];

            else if( nums1[a] < nums2[b]) a++;

            else b++;
        }
        return -1;



        

    }
};

File: 2540-minimum-common-value/2540-minimum-common-value.java
class Solution {
    public int getCommon(int[] nums1, int[] nums2) {
        int a = 0;
        int b = 0;

        while(a < nums1.length && b < nums2.length){
            if(nums1[a] == nums2[b]){
                return nums1[a];
            }else if(nums1[a] < nums2[b]){
                a++;
            }else{
                b++;
            }
        }
        return -1;
        
    }
}